src/Adapter/MailTemplate/MailTemplateTwigRenderer.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\MailTemplate;
+
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailLayoutInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailLayoutVariablesBuilderInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateRendererInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\Transformation\TransformationCollection;
+use PrestaShop\PrestaShop\Core\MailTemplate\Transformation\TransformationInterface;
+use Symfony\Component\Templating\EngineInterface;
+use Language;
```

Comment:
I'm pretty sure we have discussed somewhere about wrapping this legacy class into an adapter... do you remember about it?

Reply:
we only discussed it 1min and said that we should talk more about it..

What we could do is indeed create a minimum `LanguageInterface` which only has the current required getters and be extended later

I think it's overkill and not in this issue's scope to actually create a new core object or even an adapter I think it requires more thought before implementing the new system as `Language` is at the core of many components

But I think we could introduce a `LanuguageRepositoryInterface` and implement in an Adapter which would use the legacy class `Language` for now. Not the cleaner adapter but it keeps all the core and bundle clean without taking too much time.

What do you think?